content/chainguard/fips/not-approved-algorithms.md

```diff
@@ -0,0 +1,170 @@
+---
+title: "FIPS and not-approved MD5"
+linktitle: "FIPS and MD5"
+type: "article"
+description: "Technical deep-dive into Chainguard FIPS images access to not-approved algorithms such as MD5 and SHA1"
+date: 2025-10-28T08:00:00+00:00
+lastmod: 2025-10-28T08:00:00+00:00
+draft: false
+tags: ["FIPS", "MD5"]
+images: []
+weight: 040
+toc: true
+---
+
+## Overview
+
+FIPS cryptographic modules implement cryptographically strong protection of data at rest and in transit. NIST position on this is very [clear](https://csrc.nist.gov/projects/cryptographic-module-validation-program):
+
+> Non-validated cryptography is viewed as providing no protection to the information or data—in effect the data would be considered unprotected plaintext. If the agency specifies that the information or data be cryptographically protected, then FIPS 140-2 or FIPS 140-3 is applicable. In essence, if cryptography is required, then it must be validated. Should the cryptographic module be revoked, use of that module is no longer permitted.
+
+Given cryptographic strength and wide adoption and availability of NIST approved algorithms, many of them have been widely deployed for non-cryptographic purposes. For example many secure digests are often used as identifiers, one-way compression functions, hash maps, cyclic redundancy checks. They are used out of convenience, rather than their cryptographic security. In the following article allowed and unallowed usage is expalined, as well specific implementation guidance is provided for all currently implemented solutions.
+
+## NIST guidance on not-approved algorithm usage
+
+As part of FIPS collection of publications, NIST publishes [FIPS 140-3 Implementation Guidance](https://csrc.nist.gov/projects/cryptographic-module-validation-program/fips-140-3-ig-announcements) (FIPS 140 I.G.), the Cryptogrpahic Module Validation Programme (CMVP) is bound by resolutions and guidance published in FIPS 140-3 I.G. All sections inside that document are stable and are often referenced in the Security Policies and externally.
+
+The FIPS 140-3 I.G. 2.4.A "Definition and Use of a non-Approved Security Function" is three pages long, and must be read in conjungtion with all other NIST & ISO publications. It provides many examples, exceptions and carve outs that at times allow to use not-approved algorithms as part of higher level approved services. For example, some algorithms might not be safe to use directly, but with appropriate safeguards can be cryptographically secure. This is often the case with complex protocols such as TLS which combines cryptographic primitives in a safe way.
+
+Skipping to additional comments, let's focus on these statements (current edition, see up to date FIPS 140-3 IG for any changes).
```

Comment:
`s/see up to date/see the current/`

And maybe link to the IG again in case they somehow missed the link above.